src/key_info_managers/mod.rs

```diff
@@ -43,11 +43,23 @@ impl fmt::Display for KeyTriple {
 /// Information stored about a key
 #[derive(Serialize, Deserialize, Debug, PartialEq, Clone, Zeroize)]
 #[zeroize(drop)]
-pub struct KeyInfo {
+struct KeyInfo {
     /// Reference to a key in the Provider
-    pub id: Vec<u8>,
+    id: Vec<u8>,
     /// Attributes of a key
-    pub attributes: Attributes,
+    attributes: Attributes,
+}
```

Comment:
Hmm, does it matter whether the inner fields of a private type are public? I'm ambivalent on this, tbh, just curious what the argument was

Reply:
I'm not sure if it matters I just thought that the issue wanted all the parts of the struct to be private but I may have misunderstood. I can change this if necessary and resubmit the PR.

Reply:
I think it's good practice if they are private by default, unless needed in another module?

Reply:
Well, not sure - saying this mostly because a lot of our types don't really represent some well-defined abstract entity, they're just wrappers around things that we tend to pass around together. Like here you end up splitting the contents into `(Vec<u8>, Attributes)` at the end, so `KeyInfo` isn't really more than a tuple in disguise - don't know if hiding the contents is relevant in any way. Maybe to make it impossible to mutate? But you never pass this struct outside of the module, so you could just mutate the contents before sending them out.

Reply:
The context of making `KeyInfo` private is explained [here](https://github.com/parallaxsecond/parsec/issues/400). Basically to force people using the other methods that allow to directly get the deserialised Key ID instead of just a vector of bytes. I agree with your "tuple in disguise" point, see [this comment](https://github.com/parallaxsecond/parsec/pull/469#discussion_r667791363) where I say it would be better to remove the `Vec<u8>` or use `T` directly.

I believe even though the setters/getters are public, because `KeyInfo` is not public it means that the fields can only be accessed in this file (ie the setters/getters are not strictly needed for now).